### PR TITLE
Add PeriodicReaderWithOwnThread

### PR DESCRIPTION
**Pull request by cijothomas**
Originally part of https://github.com/open-telemetry/opentelemetry-rust/pull/2142/files, but this one leaves existing PeriodicReader as-is, and instead adds a new one `PeriodicReaderWithOwnThread`. While implementing 2142, we discovered several issues related to async runtimes + exporters requiring async runtimes(OTLPExporter) that will take some time to resolve. Hence, this is a separate reader, currently useable with stdoutexporter, etw or user-events exporter, as these exporters don't rely on any particular runtimes. Once the stability is verified, will modify this to allow usage with OTLP Exporters and eventually get merged with PeriodicReader itself.

### CODE REVIEW

File: examples/metrics-basic/src/main.rs
Comment (cijothomas) [inline examples/metrics-basic/src/main.rs:36]: modified example just to demo the working on new reader. These changes will be reverted before merging, this is only to help anyone wanting to quickly test.

Comment (cijothomas) [reply examples/metrics-basic/src/main.rs]: this is removed. Pick the changes from the first commit if one wants to try it out.

File: opentelemetry-sdk/src/metrics/mod.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:53]: Would it make sense to add this behind a feature flag to indicate to end-users that it’s still experimental ?

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:358]: We previously discussed this approach, though I don’t fully recall if we agreed on it or if we’re open to either option:
- Passing the timeout to the exporter
- Using manual polling within a loop, checking for timeout in each iteration.
If we're still considering both options, it might be best to add a TODO comment for clarity."

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:396]: Add a TODO to also call the exporter::force_flush once ollect-export is done?

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (utpilla) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:142]: I think we should use a different name. Something more intuitive to the user (component author) like: `PeriodicReaderWithBlockingExporter` or `PeriodicReaderWithSynchronousExporter` or `PeriodicReaderWithNoRuntime`

Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:142]: As mentioned in the PR desc, this will be merged into PeriodicReader itself for stable, so don't have to worry about a perfect name.
I'll move this under a feature flag as suggested here: https://github.com/open-telemetry/opentelemetry-rust/pull/2292#discussion_r1835184394

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:358]: added

File: opentelemetry-sdk/src/metrics/mod.rs
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/mod.rs:53]: added

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:253]: Perhaps hybrid approach can be one option - enforce  that the export `timeout` is configured to be less than the `interval`. And then, if the delay is slightly more than the interval, we can do next export immediately. However, if the delay is in multiple of intervals, it signifies heavy load on system, and better to skip next cycle - Just a thought, but good to discuss for further PRs.

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:577]: yes sleep will slow down the CI. Instead of sleep, we can have a  loop that periodically checks this condition as below (not syntactically correct):
```
while i.load(Ordering::Relaxed) < 5 && elapsed_time < 5*20 ms) 
{
}
```

Or better to skip this test for the CI. Not a blocker for this PR.

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:738]: nit - These 3 helper methods can be wrapped in a single method and called in all test cases to reduce redundancy

File: opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs:746]: Either skip in CI, or conditional polling. Not a blocker here though - for separate PR.

**Review by lalitb — APPROVED** [review]
LGTM with nit comments. As this is under feature flag, it can be improved further after release. Thanks for working on it.

**Review by utpilla — APPROVED** [review]
Looks good overall!

**Comment by cijothomas** [comment]
Merging to unblock release. This component is under experimental ff, and no impact on any existing code.

**Pull request merged**
